jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -760,7 +772,7 @@ protected void exitScope(Request request, Context lastContext, ClassLoader lastL
     {
         notifyExitScope(request);
         __context.set(lastContext);
-        Thread.currentThread().setContextClassLoader(lastLoader);
+        setContextClassLoader(lastLoader);
```

Comment:
We do not want to pay the cost if the the TCCL was not set, so this should be:


```suggestion
        if (_classLoader != null)
            setContextClassLoader(lastLoader);
```

Reply:
Either this or what @janbartel proposed

Reply:
Done.